llvm/test/CodeGen/AArch64/frem-power2.ll

```diff
@@ -13,31 +13,57 @@ entry:
 }
 
 define float @frem2_nsz(float %x) {
-; CHECK-LABEL: frem2_nsz:
-; CHECK:       // %bb.0: // %entry
-; CHECK-NEXT:    fmov s1, #2.00000000
-; CHECK-NEXT:    b fmodf
+; CHECK-SD-LABEL: frem2_nsz:
+; CHECK-SD:       // %bb.0: // %entry
+; CHECK-SD-NEXT:    fmov s1, #2.00000000
+; CHECK-SD-NEXT:    fdiv s2, s0, s1
```

Comment:
Is there some reason to prefer this sequence over `fmov s1, #0.5; fmul s2, s0, s1`?

Reply:
Sounds like a nice idea. It looks like there is an instcombiner for transforming that, but no equivalent in the backend at the moment. I'll try and take a look.